### PR TITLE
Upgrade to webpack@5

### PR DESCRIPTION
**Pull request by oleksandrpravosudko-okta**
## Description:

Upgrade to webpack@5 due to vulnerability in v3.12.0

* replace UglifyJsPlugin with TerserPlugin, remove RemoveStrictPlugin
* create a follow-up ticket to provide more informative bundle report 

Bundle size comparison:

|        | webpack@3           | webpack@5  |
| ------------- |:-------------:|:-----:|
| okta-sign-in.entry.js | Screenshot 2021-09-20 at 11 40 13|Screenshot 2021-09-20 at 11 41 38|
| okta-sign-in.no-polyfill.min.js |Screenshot 2021-09-20 at 11 32 12      | Screenshot 2021-09-20 at 11 33 46 |
| okta-sign-in.min.js | Screenshot 2021-09-20 at 11 35 24 | Screenshot 2021-09-20 at 11 36 44 |
| okta-sign-in-no-jquery.js |Screenshot 2021-09-20 at 11 38 16|Screenshot 2021-09-20 at 11 39 09|
| okta-sign-in.js |Screenshot 2021-09-20 at 11 43 08|Screenshot 2021-09-20 at 11 43 57|

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-333640](https://oktainc.atlassian.net/browse/OKTA-333640)

### CODE REVIEW

File: scripts/buildtools/webpack/plugins.js [hunk 1]
```diff
@@ -30,87 +26,40 @@ function devMode() {
...
 }
 
 function failOnBuildFail() {
-  return function() {
-    this.plugin('done', function(build) {
+  const FailOnBuildPlugin = function () {};
+  FailOnBuildPlugin.prototype.apply = function(compiler) {
+    compiler.hooks.done.tap('Fail on build', (stats) => {
+
       // webpack 3.x and karma-webpack combo will fail to treat seom missing assets as build failures
```
Comment (shuowu) [inline scripts/buildtools/webpack/plugins.js:40]: remove the comment?

Comment (oleksandrpravosudko-okta) [reply scripts/buildtools/webpack/plugins.js]: Updated to current webpack version - it is still relevant with webpack@5 + karma-webpack

File: package.json [hunk 2]
```diff
@@ -52,7 +52,7 @@
     "build:release": "grunt build:release",
     "build:webpack-dev": "webpack --config webpack.dev.config.js",
     "build:webpack-release": "webpack --config webpack.release.config.js",
-    "build:webpack-e2e-app": "webpack target/e2e/app/app-using-osw-entry.js target/js/app-bundle.js",
```
Comment (oleksandrpravosudko-okta) [inline package.json:55]: latest webpack-cli doesn't seem to support output filename parameter, defaulting to `main.js`

File: webpack.common.config.js [hunk 3]
```diff
@@ -81,9 +82,78 @@ module.exports = function(outputFilename) {
...
+              warnings: false
+            };
+  
+            if (sourceMap) {
+              uglifyJsOptions.sourceMap = {
+                content: sourceMap,
+              };
+            }
+  
+            return require('uglify-js').minify(file, uglifyJsOptions);
```
Comment (sherizada-okta) [inline webpack.common.config.js:153]: Why are you using Uglify within Terser?

Comment (oleksandrpravosudko-okta) [reply webpack.common.config.js:153]: there seem to be no clear way to specify license comments exceptions with Terser:
https://github.com/okta/okta-signin-widget/blob/8d9bbd5ff0e257e1391b3047750e02ee9b180c7c/scripts/buildtools/webpack/plugins.js#L64

Comment (oleksandrpravosudko-okta) [reply webpack.common.config.js]: @sherizada-okta `uglify-js` removed

**Review by aarongranick-okta — APPROVED** [review]
Amazing work! 🚀

**Pull request closed without merging**